Create Groups

Waidhoferj opened the pull request:
Fixes #138
Users can now create groups on the Group List page!

## Testing
1. Go to the Group List Page
2. Click on the plus icon in the upper right
3. Fill out the fields and press "Create Group"
4. Verify that you get a toast with an explanation and the group appears in the list.

Commit:
graphql file made

Commit:
Group creation working

gwholland3 commented on frontend/src/components/Popup/Popup.tsx:
```diff
@@ -0,0 +1,29 @@
+  onClose: () => void;
+}
+
+export default function Popup(props: PopupProps) {
```
Is this something we can insert anywhere?

Waidhoferj replied:
Yeah! The popup component should work on any screen. Just wrap it in an `AnimatePresence` and you'll get the animations too.

gwholland3 commented on frontend/src/pages/Groups/GroupPage/GroupPage.tsx:
```diff
@@ -94,8 +94,7 @@ function GroupHub({ group }: GroupHubProps) {
       push("/groups");
-    }
-    else {
+    } else {
```
Was this and other formatting stuff suggested by Prettier or something?

Waidhoferj replied:
Yes

gwholland3 reviewed (approved):
LGTM!

gwholland3 commented:
Actually one thing I just noticed: when creating the group, it seems like the description text requires a minimum of 10 characters, but you don't get an error message telling you that if you try to submit with fewer

gwholland3 commented:
And one last bug I noticed: If you leave a group that you were the last member of, it appears as though the group is deleted, both on your groups page and in the graphql interface. However, if you had copied the url of that page and try to visit it after it's deleted, you get an error instead of the "couldn't find the group you are looking for" message.

Not sure if this is related to your changes or if this should be opened as a separate issue

Commit:
updated to address bugs

Waidhoferj commented:
@gwholland3 I think the group getting deleted should be expected behavior. If no one belongs to that group, it wouldn't be reachable and we would be storing data for no reason.

I addressed the length issue and added more descriptive error handling. I don't think we need to specify length so just took it out.

gwholland3 commented:
@Waidhoferj yeah I understand that the group should be deleted when there is no one in it, but what I though was possibly a bug is that if I try to visit `/group/any_random_id_string`, it gives me a nicely formatted screen telling me that it couldn't find said group, with a go back button. 

However, if I try to visit the url of a group that was just deleted (by me being the only member and leaving it), then I get an ugly react error.

Commit:
fixed error after group deleted

Waidhoferj commented:
I also changed the catch all route from a redirect back to a normal route component. If we use the redirect component, the autologin functionality wont work because in the split second before the user is auto logged in, they are redirected.

gwholland3 reviewed (approved):
Looks good! Nice fixes

kyleshare commented:
Looks good! Only problem I noticed is that the "Artworks" metric is not being updated from 0 when I add an artwork to the group I created.

Braden50 commented:
> Looks good! Only problem I noticed is that the "Artworks" metric is not being updated from 0 when I add an artwork to the group I created.

I have an issue for this (that is assigned to me). I'm not sure if I will get it fixed in time, but I will look into it.

Braden50 reviewed (approved):
Looks beautifully simple on the front end. Nice work! Scroll bars are still many places in the app including the pop up here, but that is totally ok given our timeline.
